### DISCUSSION TITLE
How does eoreader process the read terraSAR-X data？

### DISCUSSION THREAD

**Discussion by WeiLeZhang** [sertit/eoreader#64] [Q&A] [answered]
Hello,I am using eoreader to read the TerraSAR-X product, I found that the shape of the `array ` in `prod`'s `xarray.DataArray` after reading is inconsistent with the shape of the `tif` file output by `prod.output`, as shown in the figure below. So I wonder what eoreader does with it？

The shape of the `array ` in `prod`'s `xarray.DataArray`:
![image](https://user-images.githubusercontent.com/103939997/205062870-5f4bf732-ce82-4d94-8f79-8c498ac389c9.png)

The shape of the `tif` file output by `prod.output`:
![image](https://user-images.githubusercontent.com/103939997/205063270-e1863edd-63c7-45d9-85d5-05869c7d55bf.png)

Here is my test code：
![image](https://user-images.githubusercontent.com/103939997/205065330-a656c03a-d326-4587-a713-134f56a40cf9.png)
![image](https://user-images.githubusercontent.com/103939997/205065816-f6abaaca-1e3a-4689-9ba9-cf93bd33ce8f.png)
![image](https://user-images.githubusercontent.com/103939997/205066155-74537d02-42d6-48c5-a48f-ddcd5cbfde43.png)
![image](https://user-images.githubusercontent.com/103939997/205063742-3db23ee6-89d7-4c1b-9099-a5de596b739c.png)

I am looking forward to your answer！

**Comment by remi-braun** [comment, marked as answer]
Hello,

I'm not sure I understand your issue. 😅 
You are saying that read by Python your band has a shape of (1, 9459, 7351) but when you open it elsewhere it has a shape of (21212, 16484) ?
That's strange.

Could you do a `gdalinfo` on your file and display your result ?

PS: Please note that `GEC` products have never been tested for TSX data (as seen [here](https://eoreader.readthedocs.io/en/v0.17.0/sar.html#product-type-handling)).

**Reply by remi-braun** [reply]
PS: don't hesitate to initialize `EOReader` logger to INFO at least, to see warnings and logs 😄

**Reply by WeiLeZhang** [reply]
Ok, thanks for your answer. I am checking it.

**Comment by remi-braun** [comment]
Maybe the answers in #65 can answer your questions 😄
